go-selinux/selinux_linux_test.go

```diff
@@ -16,25 +16,61 @@ func TestSetFileLabel(t *testing.T) {
 		t.Skip("SELinux not enabled, skipping.")
 	}
 
-	tmp := "selinux_test"
-	con := "system_u:object_r:bin_t:s0"
-	out, err := os.OpenFile(tmp, os.O_WRONLY|os.O_CREATE, 0)
+	const (
+		tmpFile = "selinux_test"
+		tmpLink = "selinux_test_link"
+		con     = "system_u:object_r:bin_t:s0"
+		con2    = "system_u:object_r:bin_t:s1"
+	)
+	out, err := os.OpenFile(tmpFile, os.O_WRONLY|os.O_CREATE, 0)
 	if err != nil {
-		t.Fatalf("unable to open %s: %s", tmp, err)
+		t.Fatal(err)
 	}
 	out.Close()
-	defer os.Remove(tmp)
+	defer os.Remove(tmpFile)
 
-	if err := SetFileLabel(tmp, con); err != nil {
+	if err := os.Symlink(tmpFile, tmpLink); err != nil {
+		t.Fatal(err)
+	}
+	if err := SetFileLabel(tmpLink, con); err != nil {
 		t.Fatalf("SetFileLabel failed: %s", err)
 	}
-	filelabel, err := FileLabel(tmp)
+	filelabel, err := FileLabel(tmpLink)
 	if err != nil {
 		t.Fatalf("FileLabel failed: %s", err)
 	}
-	if con != filelabel {
+	if filelabel != con {
 		t.Fatalf("FileLabel failed, returned %s expected %s", filelabel, con)
 	}
+
+	// Using LfileLabel to verify that the symlink itself is not labeled.
+	linkLabel, err := LfileLabel(tmpLink)
+	if err != nil {
+		t.Fatalf("LfileLabel failed: %s", err)
+	}
+	if linkLabel != "" {
```

Comment:
This should be linkLabel == con

Reply:
My thinking here was that no label should be set _at all_ on the symlink, correct?

Reply:
All files have labels.  Getting a file without a label on an SELinux system is difficult.